toolkit/tools/internal/sliceutils/sliceutils.go

```diff
@@ -37,11 +41,40 @@ func FindMatches(slice []string, isMatch func(string) bool) []string {
 	return result
 }
 
-// StringMatch is intended to be used with "Find" for slices of strings.
+// StringMatch is intended to be used with "Contains" and "Find" for slices of strings.
 func StringMatch(expected, given interface{}) bool {
+	if checkValid, checkResult := nilCheck(expected, given); checkValid {
+		return checkResult
+	}
+
 	return expected.(string) == given.(string)
 }
 
+// PackageVerMatch is intended to be used with "Contains" and "Find" for slices of *pkgjson.PackageVers.
+func PackageVerMatch(expected, given interface{}) bool {
```

Comment:
Huh, I'm actually surprised we don't have something like this. Could this live in pkgjson.go as a method of the packagever type maybe?

Reply:
Well this package is for working on slices but matchers can live closer to what they are supposed to match. In other words: I see it as reasonable to move it to `pkgjson.go`. I'll take a closer look in a minute.

Reply:
OK, I'll leave the matcher inside `sliceutils` for now. Moving that matcher to `pkgjson` creates a cyclic dependency, since we use the `nilCheck` function in the matcher.

Also, I want to keep all of the `*SetToSlice` functions inside `sliceutils`, so they are all inside one package, so we'd be left with a dependency on `pkgjson` anyway. Bear in mind, that this part is supposed to be temporary - once we upgrade to Goland 1.18+, I'll replace these functions with a generic `SetToSlice[T...]` and we'll be able to drop this weird dependency.

Once the generics are there, then we will no longer have `sliceutils` depend on other packages and we will be free to distribute the matchers to sit next to the types they are meant to match.